src/lib.rs

```diff
@@ -1378,6 +1382,289 @@ impl Default for Versioning {
+/// A version requirement expression, like `^1.4.163`. Equivalent to a `VersionReq` for
+/// a `Versioning` instead of a semver `Version`.
+pub struct VersioningReq {
+    /// The version requirement operation, or `None` if the operation is `*`.
+    op: Op,
+    /// The version requirement expression
+    version: Option<Versioning>,
+}
+
+impl VersioningReq {
+    /// Create a new `VersioningReq` from an operation and a version.
+    ///
+    /// # Arguments
+    ///
+    /// * `op` - The operation to use for the requirement.
+    /// * `version` - The version to use for the requirement. If `None`, the version is empty,
+    ///              which happens when the operation is `*`.
+    pub fn new(op: Op, version: Option<Versioning>) -> Self {
+        Self { op, version }
+    }
+
+    /// Return the operation of the requirement.
+    pub fn op(&self) -> Op {
+        self.op
+    }
+
+    /// Return the version of the requirement.
+    pub fn version(&self) -> Option<&Versioning> {
+        self.version.as_ref()
+    }
+
+    /// Return whether the requirement tilde-matches another requirement
+    fn matches_tilde(&self, other: &Versioning) -> bool {
+        if let Some(version) = self.version.as_ref() {
+            version_triples_match_tilde(version, other)
+        } else {
+            false
+        }
+    }
+
+    /// Return whether the requirement caret-matches another requirement
+    fn matches_caret(&self, other: &Versioning) -> bool {
+        if let Some(version) = self.version.as_ref() {
+            version_triples_match_caret(version, other)
+        } else {
+            false
+        }
+    }
+
+    /// Check if a version matches a version constraint.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use versions::{VersioningReq, Versioning};
+    /// use std::str::FromStr;
+    ///
+    /// let constraint_gt = VersioningReq::from_str(">=1.0.0").expect("Invalid req");
+    /// assert!(constraint_gt.matches(&Versioning::new("1.0.0").expect("Invalid version")));
+    /// assert!(constraint_gt.matches(&Versioning::new("1.1.0").expect("Invalid version")));
+    /// assert!(!constraint_gt.matches(&Versioning::new("0.9.0").expect("Invalid version")));
+    ///
+    /// let constraint_wild = VersioningReq::from_str("*").expect("Invalid req");
+    /// assert!(constraint_wild.matches(&Versioning::new("1.0.0").expect("Invalid version")));
+    /// assert!(constraint_wild.matches(&Versioning::new("1.1.0").expect("Invalid version")));
+    /// assert!(constraint_wild.matches(&Versioning::new("0.9.0").expect("Invalid version")));
+    ///
+    /// let constraint_eq = VersioningReq::from_str("==1.0.0").expect("Invalid req");
+    /// assert!(constraint_eq.matches(&Versioning::new("1.0.0").expect("Invalid version")));
+    /// assert!(!constraint_eq.matches(&Versioning::new("1.1.0").expect("Invalid version")));
+    /// assert!(!constraint_eq.matches(&Versioning::new("0.9.0").expect("Invalid version")));
```

Comment:
You can just `unwrap` in these doctests instead of supplying a failure reason with `expect`. It would be a bit cleaner.

Reply:
Ah, was using forbid unwrap and forgot to change it :)